mne/channels/_dig_montage_utils.py

```diff
@@ -0,0 +1,180 @@
+    with f as fid:
+        dig = _read_dig_fif(fid, tree)
+
+    # Split up the dig points by category
+    hsp = list()
+    hpi = list()
+    elp = list()
+    point_names = list()
+    fids = dict()
+    dig_ch_pos = dict()
+    for d in dig:
+        if d['kind'] == FIFF.FIFFV_POINT_CARDINAL:
+            _check_frame(d, 'head')
+            fids[_cardinal_ident_mapping[d['ident']]] = d['r']
+        elif d['kind'] == FIFF.FIFFV_POINT_HPI:
+            _check_frame(d, 'head')
+            hpi.append(d['r'])
+            elp.append(d['r'])
+            point_names.append('HPI%03d' % d['ident'])
+        elif d['kind'] == FIFF.FIFFV_POINT_EXTRA:
+            _check_frame(d, 'head')
+            hsp.append(d['r'])
+        elif d['kind'] == FIFF.FIFFV_POINT_EEG:
+            _check_frame(d, 'head')
+            dig_ch_pos['EEG%03d' % d['ident']] = d['r']
+
+    return Bunch(
+        nasion=fids['nasion'], lpa=fids['lpa'], rpa=fids['rpa'],
+        hsp=np.array(hsp) if len(hsp) else None,
+        hpi=hpi,  # XXX: why this is returned as empty list instead of None?
+        elp=np.array(elp) if len(elp) else None,
+        point_names=point_names,
+        dig_ch_pos=dig_ch_pos,
+        coord_frame='head',
+    )
+
+
+def _read_dig_montage_egi(
+        fname,
+        _scaling,
+        _all_data_kwargs_are_none,
+):
+
+    if not _all_data_kwargs_are_none:
+        raise ValueError('hsp, hpi, elp, point_names, fif must all be '
+                         'None if egi is not None')
+    _check_fname(fname, overwrite='read', must_exist=True)
+
+    root = ElementTree.parse(fname).getroot()
+    ns = root.tag[root.tag.index('{'):root.tag.index('}') + 1]
+    sensors = root.find('%ssensorLayout/%ssensors' % (ns, ns))
+    fids = dict()
+    dig_ch_pos = dict()
+
+    fid_name_map = {'Nasion': 'nasion',
+                    'Right periauricular point': 'rpa',
+                    'Left periauricular point': 'lpa'}
+
+    for s in sensors:
+        name, number, kind = s[0].text, int(s[1].text), int(s[2].text)
+        coordinates = np.array([float(s[3].text), float(s[4].text),
+                                float(s[5].text)])
+
+        coordinates *= _scaling
+
+        # EEG Channels
+        if kind == 0:
+            dig_ch_pos['EEG %03d' % number] = coordinates
+        # Reference
+        elif kind == 1:
+            dig_ch_pos['EEG %03d' %
+                       (len(dig_ch_pos.keys()) + 1)] = coordinates
+        # Fiducials
+        elif kind == 2:
+            fid_name = fid_name_map[name]
+            fids[fid_name] = coordinates
+        # Unknown
+        else:
+            warn('Unknown sensor type %s detected. Skipping sensor...'
+                 'Proceed with caution!' % kind)
+
+    return Bunch(
+        # EGI stuff
+        nasion=fids['nasion'], lpa=fids['lpa'], rpa=fids['rpa'],
+        dig_ch_pos=dig_ch_pos, coord_frame='unknown',
+        # not EGI stuff
+        hsp=None, hpi=None, elp=None, point_names=None,
```

Comment:
I would use here the `_make_dig_points` function so this returns a `dig`

Reply:
I disagree with this one. (at least for the moment) I would not return the dig. I would keep the intermediate representation and build the dig in the constructor. Like this, it is clear what every reader is using just by reading the return signature. Otherwise, I've to understand what the reader is actually doing.

Maybe at the end when everything is clear, etc but. I check in the PR all the work I did to unfold it. Returning a None is costless, unfolding what the reader does takes more time.

Reply:
next PR.